scripts/iso_mirna_quantification.py

```diff
@@ -0,0 +1,337 @@
+            "underscore and the NH value, i.e 1-2_4. Default %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+    parser.add_argument(
+        '--len',
+        help=(
+            "Include the species length in the output table."
+            "Default: %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+    parser.add_argument(
+        '--read-ids',
+        help=(
+            "Include read IDs of the alignments that are the same species in"
+            "the output table. Default: %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+
+    return parser
+
+
+def get_out_dirs(mir_list: list[str], library: str, outdir: Path) -> Optional[Path]:
+    """Get output directories.
+    
+    Create directories for each miRNA type specified in a list and return a
+    the new directories. If a miRNA type is not found, its corresponding value
+    in the returned list is None.
+
+    Args:
+        mir_list:
+            list with the desired miRNA types tables
+        library:
+            library where the species to be classified belong to
+        outdir:
+            path to the output directory
+    Returns:
+        the new directories and/or None
+    """
+    mir_out = None
+    isomir_out = None
+    iso_mirna_out = None
+
+    if 'mirna' in mir_list:
+        mir_out = outdir/f'mirna_counts_{library}'
+    if 'isomir' in mir_list:
+        isomir_out = outdir/f'isomir_counts_{library}'
+    if 'iso_mirna' in mir_list:
+        iso_mirna_out = outdir/f'iso_mirna_counts_{library}'
+
+    return mir_out, isomir_out, iso_mirna_out
+
+
+def get_contribution(aln: pysam.AlignedSegment, collapsed: bool = False, nh: bool = False) -> float:
+    """Get contribution of an alignment to the overall count.
+
+    Calculate the contribution of the alignment as the number of reads divided
+    by NH. The arguments `collapsed` and `nh` determine if these values are
+    taken from the alignment name. If both arguments are set to `True`, the
+    contribution of each alignment is computed as the number of reads/NH and
+    the values are extracted from the name. If only `collapsed` is `True`, the
+    contribution is the number of reads/NH tag. If only `nh` is `True`, the
+    contribution is 1/NH. Otherwise, the contribution is 1/NH tag. If `nh` is
+    `False` and the alignment misses the NH tag, the value is set to 1.
+
+    Args:
+        aln:
+            alignment to calculate the contribution from
+        collapsed:
+            indicates if the alignment is collapsed or not
+        nh:
+            indicates if the NH vale is found on the alignment name
+    Returns:
+        the contribution of the alignment to the overall total
+    """
+    query_id = aln.query_name
+
+    if collapsed and nh:
+        num_reads = int(query_id.split('-')[1].split('_')[0])
+        nh_value = int(query_id.split('-')[1].split('_')[1])
```

Comment:
This will fail for many query names, e.g., for `my_funny-query_name-123_4`. Use a regular expression instead. You will need to have `_NH` pinned at the end and `-COUNT` either at the end _or_ followed by `_NH`.